src/libclipper/include/clipper/datatypes.hpp

```diff
@@ -27,6 +27,7 @@ enum class RequestType {
 };
 
 size_t versioned_model_hash(const VersionedModelId &key);
+std::string versioned_model_to_str(const VersionedModelId &model);
```

Comment:
Returning const values is a bad idea (http://stackoverflow.com/a/8716406/814642). It prevents rvalue optimization which avoids a copy. But also from an API perspective helper functions should impose minimal restrictions on the calling functions. Taking const parameters when possible is good, because it allows the function to be called with either const or non-const arguments. Conversely, returning non-const values is good, because it allows the calling function to mutate the result if needed.